src/LinkDotNet.Blog.Web/RegistrationExtensions/StorageProviderExtensions.cs

```diff
@@ -49,6 +51,17 @@ public static void AddStorageProvider(this IServiceCollection services, IConfigu
         }
     }
 
+    public static void InitializeDatabase(this WebApplication app)
+    {
+        ArgumentNullException.ThrowIfNull(app);
+
+        using var scope = app.Services.CreateScope();
```

Comment:
We could also directly call the migration here?
So people will always have the latest and greatest when they start the application.

See https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/applying?tabs=dotnet-core-cli

Reply:
Do we need the scope?

If we don't register the Initialzer as scoped? Transient or even singleton might be fine as well.
It will be called only once anyway. T

Reply:
> We could also directly call the migration here? So people will always have the latest and greatest when they start the application.
> 
> See https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/applying?tabs=dotnet-core-cli

Currently, I execute the initializer at the Main, I am not sure if it is a good idea but I found it more understandable since it is executing before `app.Run()`

```
    public static void Main(string[] args)
    {
        var builder = WebApplication.CreateBuilder(args);
        RegisterServices(builder);

        var app = builder.Build();
        ConfigureApp(app);

       // HERE
        app.InitializeStorageProvider(app.Configuration);

        app.Run();
    }
```

also, I write a condition in InitializeStorageProvider to execute initialization only for SQL storage provider.